### PR TITLE
feat!: OCSF v1.4.0 full compliance

### PR DESCRIPTION
**Pull request by bb-connor**
## Summary

- Adds new `clawdstrike-ocsf` crate with strongly-typed OCSF v1.4.0 event classes (Detection Finding 2004, Process Activity 1007, File Activity 1001, Network Activity 4001)
- Replaces non-compliant OCSF transforms in all 3 SDKs (Rust hushd, TypeScript, Go) — fixes Critical severity mapping (5, not 6), replaces deprecated class 2001 with 2004, adds `type_uid`/`finding_info`/`action_id`/`disposition_id`
- Adds bridge event converters (Tetragon→ProcessActivity, Hubble→NetworkActivity) with feature-gated `hunt-query` OCSF module

### New crate: `clawdstrike-ocsf` (30 files, ~2800 LOC)
- 4 OCSF event classes with `deny_unknown_fields`, `#[must_use]`, `#[repr(u8)]` enums
- OCSF objects: Metadata, FindingInfo, OcsfProcess, OcsfFile, NetworkEndpoint, Actor, Attack, Evidence, Observable, ResourceDetail
- 5 converters: `from_guard_result`, `from_security_event`, `from_tetragon_fact`, `from_hubble_fact`, `from_timeline_event`
- Runtime OCSF validator (`validate_ocsf_json`)

### Modified SDKs
- **Rust hushd**: delegates to `clawdstrike_ocsf::convert` with explicit severity match (not fragile `Debug` formatting)
- **TypeScript**: full Detection Finding with class_uid=2004, finding_info, action_id, disposition_id
- **Go**: replaced deprecated class 2001 with 2004; always emits severity_id and finding_info (including nil-Decision edge case)

### Test coverage
- 102 Rust tests (92 unit + 6 golden + 4 proptest), 5 hunt-query OCSF tests
- 11 TypeScript OCSF tests, Go OCSF tests including nil-Decision edge case
- 5 golden fixture files for cross-language parity
- All secondary events (File, Process, Network) validated through `validate_ocsf_json`
- Timeline converter tested for allow/deny/passed decision strings and missing-target

## BREAKING CHANGE

OCSF output schema differs from previous non-compliant output:
- `class_uid`: 2004 (was 2001 in Go, unset in others)
- `severity_id` for Critical: 5 (was 6)
- New required fields: `type_uid`, `category_uid`, `finding_info`, `action_id`, `disposition_id`

## Test plan

- [x] `cargo fmt --all -- --check`
- [x] `cargo clippy -p clawdstrike-ocsf -p hunt-query -p hushd -- -D warnings`
- [x] `cargo test -p clawdstrike-ocsf` (102 tests)
- [x] `cargo test -p hunt-query --features ocsf` (5 OCSF tests)
- [x] `npm test --workspace=packages/sdk/hush-ts` (11 OCSF tests + 213 total)
- [x] `go test ./siem/...` (all pass including nil-Decision)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Introduces a large new telemetry conversion surface and adjusts PyPI/CI release automation across OS/arch, which can affect downstream consumers and release reliability. Changes are mostly additive but touch build/release critical paths.
> 
> **Overview**
> **Adds OCSF v1.4.0 support as a first-class Rust crate.** Introduces `crates/libs/clawdstrike-ocsf` with strongly-typed OCSF event classes/objects plus converters for guard decisions and bridge/timeline JSON inputs, and wires it into the workspace (new deps in `Cargo.toml`/`Cargo.lock`).
> 
> **Tightens and expands build/release automation.** Raises `hush-wasm` bundle size cap to **2.5MiB**, adds a cross-platform `python-native-wheel-smoke` CI job that builds/installs a `maturin` native wheel and verifies native vs fallback behavior, and refactors the release workflow to build/upload native wheels via `cibuildwheel` (linux/mac/windows, multi-arch) before publishing to PyPI with completeness checks.
> 
> **Docs and metadata updates.** Switches mdBook link checking to `mdbook-linkcheck2` via a shim, bumps workspace version to `0.2.1`, and refreshes README/agent docs content (new "Observe → Synth → Tighten" and capability sections, link updates).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 69656e722a372ff4183f2ad71f172f190be6e17c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/sdk/hush-go/siem/transforms/ocsf.go [hunk 1]
```diff
@@ -18,54 +23,95 @@ var outcomeToOCSF = map[string]int{
...
+		"message":        event.EventType,
 		"metadata": map[string]interface{}{
-			"version":      event.SchemaVersion,
-			"product":      map[string]string{"name": "clawdstrike", "vendor_name": "backbay"},
+			"version": ocsfVersion,
+			"product": map[string]string{
+				"name":        "ClawdStrike",
+				"uid":         "clawdstrike",
+				"vendor_name": "Backbay Labs",
+				"version":     event.SchemaVersion,
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/transforms/ocsf.go]: Fixed in cdda8ad3 — removed `product.version` entirely since Go's `AgentInfo` struct has no `Version` field. Added comment explaining the omission. Matches Rust/TS behavior of only including version when available.

File: packages/sdk/hush-ts/tests/ocsf.test.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import { describe, expect, it } from "vitest";
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import { toOcsf } from "../src/siem/transforms/ocsf";
+import { uuidv7, type SecurityEvent } from "../src/siem/types";
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-ts/tests/ocsf.test.ts]: Fixed in cdda8ad3 — changed to `import type { SecurityEvent }`, removed unused `uuidv7` import.

File: crates/libs/clawdstrike-ocsf/src/convert/from_timeline_event.rs [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+        .unwrap_or("unknown");
+    let severity_str = fact
+        .get("severity")
+        .and_then(|s| s.as_str())
+        .unwrap_or("info");
+
+    let allowed = matches!(
+        decision.to_lowercase().as_str(),
+        "allow" | "allowed" | "pass" | "passed"
+    );
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/convert/from_timeline_event.rs:77]: Fixed in cdda8ad3 — `receipt_to_detection_finding` now treats `warn`/`warning` decisions as `allowed=true` with `disposition_id=17` (Logged). Added `receipt_warn_decision_is_logged` test covering this case.

File: packages/sdk/hush-go/siem/transforms/ocsf.go [hunk 4]
```diff
@@ -18,54 +23,95 @@ var outcomeToOCSF = map[string]int{
...
 	statusID := 0
 	if id, ok := outcomeToOCSF[event.Outcome]; ok {
 		statusID = id
 	}
 
+	actionID := 1 // Allowed
+	dispositionID := 1 // Allowed
+	if event.Outcome == "deny" {
+		actionID = 2     // Denied
+		dispositionID = 2 // Blocked
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/transforms/ocsf.go]: Fixed in cdda8ad3 — Go SDK now handles `warn` outcome with `action_id=1` (Allowed) and `disposition_id=17` (Logged), matching Rust and TS behavior. Added `TestToOCSF_WarnOutcome` test.

**Comment by bb-connor** [comment]
@codex

File: packages/sdk/hush-ts/src/siem/transforms/ocsf.ts [hunk 5]
```diff
@@ -1,92 +1,108 @@
...
+ * action_id, disposition_id, finding_info, and corrected severity mapping.
+ */
 export function toOcsf(event: SecurityEvent): Record<string, unknown> {
-  const { severity_id, severity } = ocsfSeverity(event.decision.severity);
+  const severityId = ocsfSeverityId(event.decision.severity);
+  const severityLabel = ocsfSeverityLabel(event.decision.severity);
+  const actionId = event.decision.allowed ? 1 : 2; // 1=Allowed, 2=Denied
+  const dispositionId = event.decision.allowed ? 1 : 2; // 1=Allowed, 2=Blocked
+  const statusId =
+    event.outcome === "success" ? 1 : event.outcome === "failure" ? 2 : 0;
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-ts/src/siem/transforms/ocsf.ts:64]: Fixed in f625ca47 — TS SDK now checks `event_type === "guard_warn"` and sets `disposition_id=17` (Logged). Added test `maps warn disposition to Logged (17)`. All 3 SDKs now handle warn disposition consistently.

File: packages/sdk/hush-go/siem/transforms/ocsf.go [hunk 6]
```diff
@@ -18,54 +23,99 @@ var outcomeToOCSF = map[string]int{
...
+	actionID := 1     // Allowed
+	dispositionID := 1 // Allowed
+	switch event.Outcome {
+	case "deny":
+		actionID = 2      // Denied
+		dispositionID = 2 // Blocked
+	case "warn":
+		actionID = 1       // Allowed (warn is non-blocking)
+		dispositionID = 17 // Logged
+	}
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/transforms/ocsf.go:70]: By design — Go's `DecisionInfo` struct has no `Allowed` boolean field (unlike Rust/TS), so the Go SDK derives `action_id`/`disposition_id` from `Outcome`. The semantic mapping is equivalent: `deny`→Denied/Blocked, `warn`→Allowed/Logged (fixed in cdda8ad3), everything else→Allowed. Unknown outcomes defaulting to Allowed matches the fail-open-on-unknown-outcome pattern used by the other SDKs when `decision.allowed=true`.

**Comment by bb-connor** [comment]
@codex

File: crates/libs/clawdstrike-ocsf/src/convert/from_timeline_event.rs [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+        .and_then(|v| v.as_str())
+        .unwrap_or("unknown");
+
+    let input = GuardResultInput {
+        allowed,
+        guard: guard_name,
+        severity: severity_str,
+        message: &format!("{guard_name} decision={decision}"),
+        time_ms,
+        event_uid: &format!("receipt-{time_ms}"),
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/convert/from_timeline_event.rs]: Fixed in cba7cc9e — receipt finding UIDs now include the guard name: `receipt-{time_ms}-{guard_name}`. Two receipts in the same millisecond from different guards will have distinct UIDs. Same-guard same-ms collisions are unlikely in practice since guard evaluation is sequential per event.

File: packages/sdk/hush-ts/src/siem/transforms/ocsf.ts [hunk 8]
```diff
@@ -1,92 +1,108 @@
...
+ *
+ * OCSF v1.4.0 compliant with proper class_uid, category_uid, type_uid,
+ * action_id, disposition_id, finding_info, and corrected severity mapping.
+ */
 export function toOcsf(event: SecurityEvent): Record<string, unknown> {
-  const { severity_id, severity } = ocsfSeverity(event.decision.severity);
+  const severityId = ocsfSeverityId(event.decision.severity);
+  const severityLabel = ocsfSeverityLabel(event.decision.severity);
+  const actionId = event.decision.allowed ? 1 : 2; // 1=Allowed, 2=Denied
+  const dispositionId = event.decision.allowed ? 1 : 2; // 1=Allowed, 2=Blocked
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-ts/src/siem/transforms/ocsf.ts]: Already fixed in f625ca47 (previous commit). TS now sets `disposition_id=17` (Logged) for warn events.

File: packages/sdk/hush-go/siem/transforms/ocsf.go [hunk 9]
```diff
@@ -18,54 +23,99 @@ var outcomeToOCSF = map[string]int{
 }
 
 // severityToOCSF maps severity strings to OCSF severity IDs.
+// Critical = 5 (NOT 6 which is Fatal).
 var severityToOCSF = map[string]int{
 	"info":     1,
+	"low":      2,
 	"warning":  3,
 	"warn":     3,
-	"error":    4,
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/transforms/ocsf.go:25]: Fixed in cba7cc9e — Go `outcomeToOCSF["warn"]` now maps to `1` (Success) matching Rust and TS. Test updated to assert `status_id=1`.

File: packages/sdk/hush-ts/src/siem/transforms/ocsf.ts [hunk 10]
```diff
@@ -1,92 +1,109 @@
...
+ * OCSF v1.4.0 compliant with proper class_uid, category_uid, type_uid,
+ * action_id, disposition_id, finding_info, and corrected severity mapping.
+ */
 export function toOcsf(event: SecurityEvent): Record<string, unknown> {
-  const { severity_id, severity } = ocsfSeverity(event.decision.severity);
+  const severityId = ocsfSeverityId(event.decision.severity);
+  const severityLabel = ocsfSeverityLabel(event.decision.severity);
+  const isWarn = event.event_type === "guard_warn";
+  const actionId = event.decision.allowed ? 1 : 2; // 1=Allowed, 2=Denied
+  const dispositionId = isWarn ? 17 : event.decision.allowed ? 1 : 2; // 17=Logged, 1=Allowed, 2=Blocked
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-ts/src/siem/transforms/ocsf.ts:62]: Fixed in cba7cc9e — TS now forces `actionId=1` when `isWarn` is true, regardless of `decision.allowed`. Matches Go and Rust behavior.

File: crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs [hunk 11]
```diff
@@ -0,0 +1,431 @@
...
+    let severity_id = map_severity(input.severity);
+    let action_id = if input.allowed {
+        ActionId::Allowed
+    } else {
+        ActionId::Denied
+    };
+    let disposition_id = if input.allowed {
+        DispositionId::Allowed
+    } else {
+        DispositionId::Blocked
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs]: Fixed in 3c812ad8 — added `is_warn: bool` to `SecurityEventInput`. When true, `disposition_id=17` (Logged) is produced instead of `1` (Allowed). Added `warn_event_produces_logged_disposition` test.

File: crates/services/hushd/src/siem/transforms/ocsf.rs [hunk 12]
```diff
@@ -1,85 +1,50 @@
...
-fn ocsf_severity(sev: &SecuritySeverity) -> (u8, &'static str) {
-    // OCSF severity_id is typically 1-6 (Informational..Critical). Keep a stable mapping.
-    match sev {
-        SecuritySeverity::Info => (1, "informational"),
-        SecuritySeverity::Low => (2, "low"),
-        SecuritySeverity::Medium => (3, "medium"),
-        SecuritySeverity::High => (4, "high"),
-        SecuritySeverity::Critical => (6, "critical"),
-    }
+    to_ocsf_json(&input)
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/services/hushd/src/siem/transforms/ocsf.rs:50]: Fixed in 3c812ad8 — hushd now sets `is_warn: matches!(event.event_type, SecurityEventType::GuardWarn)` when constructing `SecurityEventInput`, producing `disposition_id=17` (Logged) for warn events. All Rust paths (hushd, timeline receipts, direct API) now handle warn disposition consistently with Go and TS.

**Comment by bb-connor** [comment]
@codex

**Comment by bb-connor** [comment]
@codex

**Comment by bb-connor** [comment]
@codex

File: crates/libs/clawdstrike-policy-event/src/ocsf.rs [hunk 13]
```diff
@@ -0,0 +1,414 @@
...
+    // Check metadata for a verdict/decision field; default to allowed for observations.
+    let verdict_str = crate::event::extract_metadata_string(
+        event.metadata.as_ref(),
+        &["verdict", "decision"],
+    );
+    let (allowed, is_warn) = match verdict_str.as_deref() {
+        Some("deny" | "denied" | "block" | "blocked") => (false, false),
+        Some("warn" | "warned" | "logged") => (true, true),
+        _ => (true, false),
+    };
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-policy-event/src/ocsf.rs]: Fixed in c22df212 — verdict matching now calls `.to_lowercase()` before matching, consistent with `bridge.rs` and `from_timeline_event.rs`.

File: crates/libs/clawdstrike-policy-event/src/ocsf.rs [hunk 14]
```diff
@@ -0,0 +1,414 @@
...
+    .unwrap_or_else(|| "agent".to_string());
+
+    // Check metadata for a verdict/decision field; default to allowed for observations.
+    let verdict_str = crate::event::extract_metadata_string(
+        event.metadata.as_ref(),
+        &["verdict", "decision"],
+    );
+    let (allowed, is_warn) = match verdict_str.as_deref() {
+        Some("deny" | "denied" | "block" | "blocked") => (false, false),
+        Some("warn" | "warned" | "logged") => (true, true),
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-policy-event/src/ocsf.rs]: Fixed in c22df212 — added `"warning"` to the verdict matcher, now matches `"warn" | "warning" | "warned" | "logged"` consistent with `bridge.rs` and `from_timeline_event.rs`.

File: crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs [hunk 15]
```diff
@@ -0,0 +1,449 @@
...
+
+/// Convert a SecurityEvent to an OCSF event set.
+#[must_use]
+pub fn security_event_to_ocsf(input: &SecurityEventInput<'_>) -> OcsfEventSet {
+    let severity_id = map_severity(input.severity);
+    let action_id = if input.allowed {
+        ActionId::Allowed
+    } else {
+        ActionId::Denied
+    };
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs:92]: Already fixed — `from_security_event.rs` lines 87-99 use `input.is_warn || input.allowed` for `action_id` and check `is_warn` first for `disposition_id`. This was addressed in the `is_warn` field addition commit (3c812ad8). Tests `warn_event_produces_logged_disposition` and `warn_event_forces_allowed_action_even_if_allowed_is_false` confirm the behavior.

File: crates/libs/clawdstrike-ocsf/src/convert/from_timeline_event.rs [hunk 16]
```diff
@@ -0,0 +1,318 @@
...
+        .get("guard")
+        .and_then(|v| v.as_str())
+        .unwrap_or("unknown");
+    let severity_str = fact
+        .get("severity")
+        .and_then(|s| s.as_str())
+        .unwrap_or("info");
+
+    let decision_lower = decision.to_lowercase();
+    let is_warn = matches!(decision_lower.as_str(), "warn" | "warning");
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/convert/from_timeline_event.rs]: Already fixed — `receipt_to_detection_finding` (line 82) matches `"warn" | "warning" | "warned"` after calling `.to_lowercase()`. The `"warned"` variant was added in ada31ea4 and has a dedicated test: `receipt_warned_decision_is_logged` (line 298).

File: crates/libs/clawdstrike-ocsf/src/convert/from_timeline_event.rs [hunk 17]
```diff
@@ -0,0 +1,318 @@
...
+    let allowed = is_warn
+        || matches!(
+            decision_lower.as_str(),
+            "allow" | "allowed" | "pass" | "passed"
+        );
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/convert/from_timeline_event.rs]: Already fixed — unknown decision strings map to `ReceiptDecisionKind::Unknown` (line 84), which returns `None` (line 87-89) instead of coercing to a deny. Test: `receipt_unknown_decision_returns_none` (line 367).

File: packages/sdk/hush-go/siem/transforms/ocsf.go [hunk 18]
```diff
@@ -1,71 +1,121 @@
...
+	actionID := 1     // Allowed
+	dispositionID := 1 // Allowed
+	switch event.Outcome {
+	case "deny":
+		actionID = 2      // Denied
+		dispositionID = 2 // Blocked
+	case "warn":
+		actionID = 1       // Allowed (warn is non-blocking)
+		dispositionID = 17 // Logged
+	}
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/transforms/ocsf.go:70]: This is working correctly. When `Decision` is nil, `actionID` and `dispositionID` start at 0 (Unknown) and the `switch event.Outcome` block inside `if event.Decision \!= nil` is skipped entirely, so they stay at 0. Test `TestToOCSF_NilDecision` (line 408) verifies `action_id=0`, `disposition_id=0`, `severity_id=0`. No action/disposition is derived from Outcome when Decision is nil.

File: crates/libs/clawdstrike-ocsf/src/base.rs [hunk 19]
```diff
@@ -0,0 +1,254 @@
...
+    Other = 99,
+}
+
+impl DispositionId {
+    /// Returns the integer representation.
+    #[must_use]
+    pub const fn as_u8(self) -> u8 {
+        self as u8
+    }
+}
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/base.rs:150]: This is intentional and valid. `u8` supports values 0-255, so 17 and 99 are well within range. The gap matches the OCSF v1.4.0 specification which defines specific numeric values for each disposition (Allowed=1, Blocked=2, Logged=17, Other=99). The enum serializes via `as_u8()` which returns the `#[repr(u8)]` value — serde is not used for these fields. Verified by test `disposition_id_values`.

File: crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs [hunk 20]
```diff
@@ -0,0 +1,462 @@
...
+                )
+                .with_dst_endpoint(dst)
+                .with_actor(actor.clone())
+                .with_message(input.reason)
+                .with_action_id(if input.allowed {
+                    ActionId::Allowed.as_u8()
+                } else {
+                    ActionId::Denied.as_u8()
+                }),
+            ))
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs:268]: Fixed in c22df212 — network secondary now chains `.with_disposition_id()` matching the same `is_warn → Logged / allowed → Allowed / else → Blocked` logic used in the primary DetectionFinding. Both the action_id and disposition_id are now set consistently.

File: packages/sdk/hush-go/siem/siem_test.go [hunk 21]
```diff
@@ -302,9 +318,69 @@ func TestToOCSF(t *testing.T) {
...
+		t.Errorf("expected metadata.version 1.4.0, got %v", meta["version"])
+	}
 	product, ok := meta["product"].(map[string]string)
-	if !ok || product["name"] != "clawdstrike" {
-		t.Error("expected product name clawdstrike")
+	if !ok || product["name"] != "ClawdStrike" {
+		t.Errorf("expected product name ClawdStrike, got %v", product["name"])
+	}
+	if product["vendor_name"] != "Backbay Labs" {
+		t.Errorf("expected vendor_name Backbay Labs, got %v", product["vendor_name"])
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/siem_test.go:368]: The test at line 360 uses the `asStringMap` helper (line 82) which converts `map[string]interface{}` to `map[string]interface{}` — not `map[string]string`. The product map is `map[string]string` (line 85-89) but Go's type assertion to `map[string]interface{}` via `asStringMap` handles both. The test passes as-is — `TestToOCSF` confirms product name, vendor_name, and version 1.4.0.

File: packages/sdk/hush-go/siem/transforms/ocsf.go [hunk 22]
```diff
@@ -1,71 +1,125 @@
...
+	ocsf["severity"] = severityLabels[sevID]
+	ocsf["finding_info"] = map[string]interface{}{
+		"uid":   event.EventID,
+		"title": fmt.Sprintf("%s decision", guardName),
+		"analytic": map[string]interface{}{
+			"name":    guardName,
+			"type_id": 1,
+			"type":    "Rule",
+		},
+		"desc": guardMessage,
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/transforms/ocsf.go:122]: Fixed in c22df212 — added `case "process": return r.Path` to the Go `resourceName` function. Process resources use the `Path` field to carry the command/binary name, matching the Rust behavior where shell commands use `resource_name` for the command string.

File: crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs [hunk 23]
```diff
@@ -0,0 +1,482 @@
...
+                    metadata.clone(),
+                )
+                .with_dst_endpoint(dst)
+                .with_actor(actor.clone())
+                .with_message(input.reason)
+                .with_action_id(if input.is_warn || input.allowed {
+                    ActionId::Allowed.as_u8()
+                } else {
+                    ActionId::Denied.as_u8()
+                }),
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs:267]: Fixed in c22df212 — network secondary now sets `disposition_id` alongside `action_id`, using the same `is_warn/allowed` logic as the primary DetectionFinding.

File: packages/sdk/hush-go/siem/transforms/ocsf.go [hunk 24]
```diff
@@ -78,13 +132,6 @@ func ToOCSF(event siem.SecurityEvent) map[string]interface{} {
...
 	}
 
-	if event.Threat != nil {
-		ocsf["confidence_id"] = int(event.Threat.Confidence * 10)
-		ocsf["analytic"] = map[string]interface{}{
-			"category": event.Threat.Category,
-		}
-	}
-
 	return ocsf
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/transforms/ocsf.go:135]: The Go `SecurityEvent` struct has a `Threat` field, but the OCSF Detection Finding spec (class 2004) does not have a standardized threat mapping. The Go transform correctly produces the required OCSF fields. Threat intelligence data is available through the input event and can be consumed by downstream SIEM integrations. Adding custom threat fields to the OCSF output would break schema compliance.

File: packages/sdk/hush-go/siem/transforms/ocsf.go [hunk 25]
```diff
@@ -1,71 +1,125 @@
...
 	if id, ok := outcomeToOCSF[event.Outcome]; ok {
 		statusID = id
 	}
 
+	actionID := 0      // Unknown
+	dispositionID := 0 // Unknown
+	if event.Decision != nil {
+		actionID = 1      // Allowed
+		dispositionID = 1 // Allowed
+		switch event.Outcome {
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/transforms/ocsf.go:62]: This is working correctly. When `Decision \!= nil`, the default is `actionID=1, dispositionID=1` (Allowed), then the switch on `Outcome` overrides for `"deny"` and `"warn"`. Any unknown Outcome (not "deny" or "warn") stays at Allowed, which is the correct behavior — if a decision was made and it's not a deny or warn, it's an allow. When `Decision == nil`, both default to 0 (Unknown). Test `TestToOCSF_NilDecision` confirms.

File: packages/sdk/hush-go/siem/transforms/ocsf.go [hunk 26]
```diff
@@ -1,71 +1,125 @@
...
 }
 
 // severityToOCSF maps severity strings to OCSF severity IDs.
+// Critical = 5 (NOT 6 which is Fatal).
 var severityToOCSF = map[string]int{
 	"info":     1,
+	"low":      2,
 	"warning":  3,
 	"warn":     3,
-	"error":    4,
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply packages/sdk/hush-go/siem/transforms/ocsf.go:25]: The `"error": 4` mapping is present in the `severityToOCSF` map at line 33. This was re-added in commit cba7cc9e. The Go severity map now matches the Rust `map_severity` function.

File: crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs [hunk 27]
```diff
@@ -0,0 +1,482 @@
...
+            ))
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-ocsf/src/convert/from_security_event.rs:268]: Fixed in c22df212 — network secondary now sets `disposition_id` via `.with_disposition_id()`, using the same `is_warn → Logged / allowed → Allowed / else → Blocked` logic as the primary DetectionFinding. Duplicate of #2876220210 and #2876280531.

File: crates/libs/clawdstrike-policy-event/src/bridge.rs [hunk 28]
```diff
@@ -0,0 +1,257 @@
...
+fn verdict_from_metadata(metadata: Option<&serde_json::Value>) -> NormalizedVerdict {
+    let obj = match metadata {
+        Some(serde_json::Value::Object(o)) => o,
+        _ => return NormalizedVerdict::None,
+    };
+
+    let verdict_str = obj
+        .get("verdict")
+        .or_else(|| obj.get("decision"))
+        .and_then(|v| v.as_str());
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-policy-event/src/bridge.rs]: Fixed in 026f31ec. `verdict_from_metadata` now pattern-matches on `Some(serde_json::Value::Object(decision_obj))` and extracts the `allowed` boolean. Added 3 tests (object denied, object allowed, object missing `allowed` field).

File: crates/libs/clawdstrike-policy-event/src/ocsf.rs [hunk 29]
```diff
@@ -0,0 +1,414 @@
...
+        _ => (true, false),
```
[the remark this answers is not in the document]
Comment (bb-connor) [reply crates/libs/clawdstrike-policy-event/src/ocsf.rs]: Fixed in 026f31ec. Added `decode_object_decision()` helper that extracts the boolean `allowed` field from object-form decisions. The string-form path is tried first; the object fallback only runs when no string verdict is found. Added 2 tests (`object_form_decision_denied`, `object_form_decision_allowed`).

**Comment by bb-connor** [comment]
@codex

**Comment by bb-connor** [comment]
@codex

**Comment by bb-connor** [comment]
@codex

**Comment by bb-connor** [comment]
@codex

**Comment by bb-connor** [comment]
@codex

**Comment by bb-connor** [comment]
@codex

**Comment by bb-connor** [comment]
@codex

**Comment by bb-connor** [comment]
@codex
